packages/ui/Scrollable/Scrollbar.js

```diff
@@ -29,165 +37,136 @@ const setCSSVariable = (element, variable, value) => {
-		 *
-		 * * `scrollbar` - The scrollbar component class
-		 *
-		 * @type {Object}
-		 * @public
-		 */
-		css: PropTypes.object,
-
-		/**
-		 * If `true`, the scrollbar will be oriented vertically.
-		 *
-		 * @type {Boolean}
-		 * @default true
-		 * @public
-		 */
-		vertical: PropTypes.bool
-	}
-
-	static defaultProps = {
-		corner: false,
-		css: componentCss,
-		vertical: true
-	}
-
-	constructor (props) {
-		super(props);
-
-		this.containerRef = React.createRef();
-		this.thumbRef = React.createRef();
-	}
-
-	componentDidMount () {
-		this.calculateMetrics();
-	}
-
-	componentDidUpdate () {
-		this.calculateMetrics();
-	}
-
-	componentWillUnmount () {
-		this.hideThumbJob.stop();
-	}
-
-	minThumbSizeRatio = 0
-	ignoreMode = false
-
-	update = (bounds) => {
-		const
-			{vertical} = this.props,
-			{clientWidth, clientHeight, scrollWidth, scrollHeight, scrollLeft, scrollTop} = bounds,
-			clientSize = vertical ? clientHeight : clientWidth,
-			scrollSize = vertical ? scrollHeight : scrollWidth,
-			scrollOrigin = vertical ? scrollTop : scrollLeft,
-
-			thumbSizeRatioBase = (clientSize / scrollSize),
-			scrollThumbPositionRatio = (scrollOrigin / (scrollSize - clientSize)),
-			scrollThumbSizeRatio = Math.max(this.minThumbSizeRatio, Math.min(1, thumbSizeRatioBase));
-
-		setCSSVariable(this.thumbRef.current, '--scrollbar-size-ratio', scrollThumbSizeRatio);
-		setCSSVariable(this.thumbRef.current, '--scrollbar-progress-ratio', scrollThumbPositionRatio);
-	}
-
-	showThumb = () => {
-		this.hideThumbJob.stop();
-		ReactDOM.findDOMNode(this.thumbRef.current).classList.add(this.props.css.thumbShown); // eslint-disable-line react/no-find-dom-node
-	}
-
-	startHidingThumb = () => {
-		this.hideThumbJob.start();
-	}
-
-	hideThumb = () => {
-		ReactDOM.findDOMNode(this.thumbRef.current).classList.remove(this.props.css.thumbShown); // eslint-disable-line react/no-find-dom-node
-	}
-
-	hideThumbJob = new Job(this.hideThumb.bind(this), thumbHidingDelay);
-
-	calculateMetrics = () => {
-		const primaryDimenstion = this.props.vertical ? 'clientHeight' : 'clientWidth';
-		let trackSize;
+const ScrollbarBase = memo(forwardRef((props, ref) => {
+	// Refs
+	const containerRef = useRef();
+	const thumbRef = useRef();
+	const hideThumbJob = useRef(null);
+	// Render
+	const
+		{childRenderer, className, corner, css, vertical, ...rest} = props,
+		containerClassName = classNames(
+			className,
+			css.scrollbar,
+			corner && css.corner,
+			vertical ? css.vertical : css.horizontal
+		);
 
-		if (this.props.clientSize) {
-			trackSize = this.props.clientSize[primaryDimenstion];
-		} else {
-			trackSize = this.containerRef.current[primaryDimenstion];
+	delete rest.clientSize;
+
+	hideThumbJob.current = hideThumbJob.current || new Job(hideThumb, thumbHidingDelay);
+
+	function hideThumb () {
+		removeClass(thumbRef.current, css.thumbShown);
+	}
+
+	useEffect(() => {
+		const job = hideThumbJob.current;
+		return () => {
+			job.stop();
+		};
+	}, []);
```

Comment:
Where do we use the `job` variable?

Reply:
I will remove this. (a missing piece from my clean up).